.github/actions/install-go-and-dependencies/action.yml

```diff
@@ -0,0 +1,18 @@
+name: InstallGo 
+description: 'Installs Go Downloads'
+runs:
+  using: "composite"
+  steps:
```

Comment:
You should audit existing workflow files and replace setup-go-and-cache with yours. e.g. https://github.com/awslabs/kubernetes-iteration-toolkit/blob/06baa4e9112320d2dcbc6c25e48ac75a1b6151cd/.github/workflows/build.yaml#L13-L24

Reply:
updating with latest versions.

Reply:
What do you mean?

Reply:
This action.yml we can use it for later workflows for provisioning the eks guest cluster and also we can update the existing ones in our workflows